src/main/java/org/aksw/simba/squirrel/sink/impl/sparql/QueryGenerator.java

```diff
@@ -7,51 +7,58 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 /**
  * This class is used to provides templates for basic SPARQL commands needed in this project.
  */
 public class QueryGenerator {
 
+    /**
+     * The instance of the class QueryGenerator.
+     */
     private static final QueryGenerator instance = new QueryGenerator();
     private static final Logger LOGGER = LoggerFactory.getLogger(QueryGenerator.class);
 
-
     private QueryGenerator() {
     }
 
-    public static QueryGenerator getInstance() {
-        return instance;
-    }
-
     /**
-     * Return an Add Query for the given graph id and triple.
+     * Getter for {@link #instance}.
      *
-     * @param graphIdentifier the given graph id.
-     * @param triples         The given triple.
-     * @return The generated query.
+     * @return instannce of the class.
      */
-    public String getAddQuery(String graphIdentifier, List<Triple> triples) {
-        String strQuery = "";
-
-        strQuery += "INSERT DATA { GRAPH <" + graphIdentifier + "> { ";
-        for (Triple triple : triples) {
-            strQuery += "<" + triple.getSubject() + "> <" + triple.getPredicate() + "> <" + triple.getObject() + "> . ";
-        }
-        strQuery += "} }";
-        return strQuery;
+    public static QueryGenerator getInstance() {
+        return instance;
     }
 
     /**
-     * Return an Add Query for a given uri and a triple.
-     *
-     * @param uri     The given Uri.
-     * @param triples The given triple.
+     *Return an Add Query for the given uri and its triples.
+     * @param uri the uri where the triples found.
+     * @param listBufferedTriples the given list of triples.
      * @return The generated query.
      */
-    public String getAddQuery(CrawleableUri uri, List<Triple> triples) {
-        return getAddQuery(uri.getUri().toString(), triples);
+    public String getAddQuery(CrawleableUri uri, ConcurrentLinkedQueue<Triple> listBufferedTriples) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("INSERT DATA { Graph <");
+        stringBuilder.append("0430");
```

Comment:
I am not sure why you want to add `0430` before the URI of the graph. I think URIs are not even allowed to start with numbers. :wink: